client/packages/common/src/hooks/useConfirmOnLeaving/useConfirmOnLeaving.ts

```diff
@@ -1,29 +1,26 @@
-import { useCallback, useContext, useEffect } from 'react';
+import { useCallback, useEffect, useState } from 'react';
 import { useBeforeUnload, useBlocker } from 'react-router-dom';
 import { create } from 'zustand';
 import { useTranslation } from '@common/intl';
-import { ConfirmationModalContext, Location } from '@openmsupply-client/common';
+import { Location, useConfirmationModal } from '@openmsupply-client/common';
 
 /**
  * useConfirmOnLeaving is a hook that will prompt the user if they try to
  * navigate away from, or refresh the page, when there are unsaved changes.
  */
-export const useConfirmOnLeaving = (
-  key: string,
-  customCheck?: (currentLocation: Location, nextLocation: Location) => boolean
-) => {
+export const useConfirmOnLeaving = (key: string, options?: BlockerOptions) => {
```

Comment:
Heh sorry more changes to this hook - updated from passing in just a `customCheck` to `options`, so we can also provide a custom confirmation modal to open instead of the generic `are you sure`